synapse/handlers/room.py

```diff
@@ -455,8 +457,24 @@ def filter_evts(events):
         else:
             last_event_id = event_id
 
+        types = None
+        filtered_types = None
+        if event_filter and event_filter.lazy_load_members():
+            members = {}
```

Comment:
hrm, can we use a set rather than a dict here?

```
members = set()
# ...
members.add(ev.sender)
```
or even:
```
members = set(ev.sender for ev in itertools.chain(
        results["events_before"],
        (results["event"],),
        results["events_after"],
))
```